dockerfile/rust/Dockerfile

```diff
@@ -18,23 +18,30 @@ WORKDIR /build/${GITHUB_REPO}
 
 RUN git checkout ${VERSION}
 
-RUN cargo fetch
```

Comment:
Sorry I missed this the first time reviewing. Instead of removing this, can we run this if BUILD_TARGET is not empty?

Reply:
My bad. Great catch!
@agouin if you think it's better to cache this, I can add it to its own line.

Reply:
Yes it would be great to keep caching for this step if the build target is present

Reply:
Done.